jenkins-master/jobs/scripts/workspace/runtests.py

```diff
@@ -4,231 +4,137 @@
+        elif args.type == 'functional':
+            command.extend(['--gecko-log', settings['logs']['gecko.log']])
 
-        elif options.type == 'functional':
             manifests = [firefox_puppeteer.manifest, firefox_ui_tests.manifest_functional]
             command.extend(manifests)
 
-        print('Calling command to execute tests: %s' % command)
-        failed = False
+        retval = 0
 
-        try:
-            subprocess.check_call(command)
-        except:
-            failed = True
-
-            # Only for failing update tests add the HTTP.log as artifact
-            http_log_url = self.upload_s3('http.log')
-            if http_log_url:
-                job.add_details(title='http.log',
-                                value=http_log_url,
-                                content_type='link',
-                                url=http_log_url)
-        finally:
-            if job and th:
-                gecko_log_url = self.upload_s3('gecko.log')
-                if gecko_log_url:
-                    job.add_details(title='gecko.log',
-                                    value=gecko_log_url,
-                                    content_type='link',
-                                    url=gecko_log_url)
-
-                tbpl_log_url = self.upload_s3('tbpl.log')
-                if tbpl_log_url:
-                    parser = JobResultParser('tbpl.log')
-                    job.add_log_reference(os.path.basename('tbpl.log'), tbpl_log_url,
-                                          parse_status='parsed')
-                    job.add_artifact(name='text_log_summary',
-                                     artifact_type='json',
-                                     blob={'step_data': parser.failures_as_json(),
-                                           'logurl': tbpl_log_url})
-
-                job.completed(result='testfailed' if failed else 'success')
-                th.submit_results(job)
-
-    def upload_s3(self, path):
-        if not self.s3_bucket:
-            return None
+        print('Calling command to execute tests: {}'.format(command))
+        retval = subprocess.call(command)
 
-        try:
-            remote_filename = '%s_%s' % (str(uuid.uuid4()), os.path.basename(path))
-            return self.s3_bucket.upload(path, remote_filename)
-        except Exception as e:
-            print 'Failure uploading "%s" to S3: %s' % (path, str(e))
-            return None
+        # Save exit code into file for further processing in report submission
+        with file('retval.txt', 'w') as f:
+            f.write(str(retval))
+
+        # Delete http.log if tests were passing
+        if not retval and settings['logs'].get('http.log'):
+            shutil.rmtree(settings['logs']['http.log'], ignore_errors=True)
+
+
+class JenkinsDefaultValueAction(argparse.Action):
+    """Jenkins cannot handle optional arguments to a CLI script so we pass in
+    a 'None' value by default.
```

Comment:
So the situation here is a combination of parameters from the parametrized plugin and executing a command via the shell in a build step. The parameters for the job contain a value and not a name + value. That means when we call the shell command we always have to pass the name and value and therefore need the name hard-coded in the command line. Best would be if the default of a job parameter is set, that this option is not passed to the shell script. But I don't see how this can be done in Jenkins.